src/09-led-compass/solution-1.md

```diff
@@ -51,21 +44,26 @@ fn main() -> ! {
     let mut sensor = Lsm303agr::new_with_i2c(i2c);
     sensor.init().unwrap();
     sensor.set_mag_odr(MagOutputDataRate::Hz10).unwrap();
+    sensor.set_accel_odr(AccelOutputDataRate::Hz10).unwrap();
     let mut sensor = sensor.into_mag_continuous().ok().unwrap();
 
+    let calibration = calc_calibration(&mut sensor, &mut display, &mut timer);
+    rprintln!("Calibration: {:?}", calibration);
+    rprintln!("Calibration done, entering busy loop");
     loop {
-        while !sensor.mag_status().unwrap().xyz_new_data  {}
-        let data = sensor.mag_data().unwrap();
+        while !sensor.mag_status().unwrap().xyz_new_data {}
+        let mut data = sensor.mag_data().unwrap();
+        data = calibrated_measurement(data, &calibration);
 
         let dir = match (data.x > 0, data.y > 0) {
             // Quadrant I
-            (true, true) => Direction::SouthEast,
+            (true, true) => Direction::NorthEast,
```

Comment:
Signs still inverted? may be worth explaining the coordinate system?

Reply:
Yeah if I understood the microbit code that is being used for calibration and applying the calibration correctly they basically convert the values into a coordinate system where the pos. Y axis is north and the pos X axis is east. Like you would more or less intuitively assume considering the concept of north as "up" and east as "right".

Reply:
My thought when writing this was basically that they should explore the coordinate system themselves (and since it looksl ike youd more or less intuitively assume it is easy to do so) but we can of course add a description.

Reply:
Ah, I see. We can leave it like this then.